network/api/v1/client.go

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (C) 2020. Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package v1
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/nuts-foundation/nuts-node/network/rfc004"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+)
+
+// HTTPClient holds the server address and other basic settings for the http client
+type HTTPClient struct {
+	ServerAddress string
+	Timeout       time.Duration
+}
+
+// CreateDocument is not supported for HTTP clients
+func (hb HTTPClient) CreateDocument(payloadType string, payload []byte, signingKeyID string, attachKey bool, timestamp time.Time, fieldsOpts ...rfc004.FieldOpt) (rfc004.Document, error) {
+	panic("not supported")
```

Comment:
HTTPClient implements some interface, but since we do no longer use a `client` interface between engines, the ClientAPI should only be a concrete implementation.